### DISCUSSION TITLE
How To Display Image Files In a Slint-based Application UI?

### DISCUSSION THREAD

**Discussion by PackMeister** [slint-ui/slint#6592] [Q&A]
We're developing a Rust/Slint/SQLite-based geneaology application for Linux desktops (Debian and NixOS) that needs to upload and display image files (png, jpeg, gif, tiff, etc) associated with a given ancestor or relative.

We can successfully upload our image files but are struggling to find a way to display them from our Slint UI.

Does Slint lack the native ability to render any image formats?

If so, are there any recommended Rust-based and Slint-compatible rendering libraries we could use to display images?

Any documentation, examples of Slint image rendering, links or tutorials on this topic would be much appreciated.

**Comment by Vadoola** [comment]
It's not clear to me from your comment exactly what you are trying that's failing.
Slint does have an Image component: https://docs.slint.dev/latest/docs/slint/src/language/builtins/elements#image

While none of my projects have used an image 'uploaded' to it from the user, in one of my projects I modify and reload the image based on theming.

Here is the link to the Rust API for the image component, that goes into more depth, maybe this will point you in the right direction?: https://docs.slint.dev/latest/docs/rust/slint/struct.Image

**Comment by hunger** [comment]
Which image formats are you using?

Slint uses the `image` crate with default features. That means

```
"avif", "bmp", "dds", "exr", "ff", "gif", "hdr", "ico", "jpeg", "png", "pnm", "qoi", "tga", "tiff", "webp"
```

should all be enabled. If you need more than that, you should also be able to pass more features to the image crate.

**Reply by ogoffart** [reply]
Note that by default with Rust, only png and jpeg are enabled:
https://github.com/slint-ui/slint/blob/c52fdd71bc2af4f8fe3b2033c0c5534efa9840f6/Cargo.toml#L144

So extra format need to be enabled by enabling feature on the image 0.24 crate.

**Reply by hunger** [reply]
True, but we tend to put the `default` back into the `features` when we use the image crate in e.g. the compiler. That's why I listed the complete set.

**Reply by ogoffart** [reply]
But we do not enable image default feature it in Slint runtime library by default, which is used to load images at runtime for programs using rust API (without EmbedForSoftwareRenderer flag).
